### ISSUE TITLE
NullPointerException evaluating calva.autoEvaluateCode.onConnect.cljs when using Krell

### ISSUE THREAD

**Issue by xavi** [BetterThanTomorrow/calva#2372]
I run "Calva: Start a Project REPL and Connect (aka Jack-In)" with "deps.edn + Krell" ( https://calva.io/krell/#starting-the-krell-clojurescript-repl ).

This works and Calva shows...

```
clj꞉user꞉> 
; Creating cljs repl session...
; Connecting cljs repl: deps.edn + Krell...
;   The Calva Connection Log might have more connection progress information.
```

then I select "React Native: Run iOS on Simulator", the simulator opens and starts the app, and then Calva's output window shows...

```
; Connected session: cljs
; TIPS: You can choose which REPL to use (clj or cljs):
;    *Calva: Toggle REPL connection*
;    (There is a button in the status bar for this)
; Evaluating code from settings: 'calva.autoEvaluateCode.onConnect.cljs'
; Unexpected error (NullPointerException) compiling at (REPL:1).
; Cannot invoke "Object.hashCode()" because "key" is null
cljs꞉user꞉>  ; Use `alt+enter` to evaluate
; Jack-in done.
cljs꞉user꞉> 
```

Why that NullPointerException?

I tried deleting the `calva.autoEvaluateCode` entry in `~/Library/Application\ Support/Code/User/settings.json` , but the error doesn't go away.

```json
"calva.autoEvaluateCode": {
    "onConnect": {
      "clj": "(when-let [requires (resolve 'clojure.main/repl-requires)] (clojure.core/apply clojure.core/require @requires))",
      "cljs": "(try (require '[cljs.repl :refer [apropos dir doc find-doc print-doc pst source]]) (catch :default e (js/console.warn \"Failed to require cljs.repl utilities:\" (.-message e))))"
    },
    "onFileLoaded": {
      "clj": null,
      "cljs": null
    }
}
```

On the other hand, I can run the onConnect command manually in the REPL with no error...
```cljs
(try (require '[cljs.repl :refer [apropos dir doc find-doc print-doc pst source]]) (catch :default e (js/console.warn "Failed to require cljs.repl utilities:" (.-message e))))
```

Any idea on why it fails with NullPointerException when it's autorun?

**Comment by PEZ** [comment]
Thanks for reporting! I actually learnt today that we shouldn't be evaluating that require in a `try` because in ClojureScript `require` only works from the top level. So it could be that, even though you managed to evaluate it in the REPL. During my experiments today I had some quite different results and I am thinking the behaviour is a bit undefined.


Deleting the entry makes Calva use the default, so that will keep evaluating that `try` expression. Can you try with setting the `"cljs"` entry there to `null` and see if that helps?

**Comment by xavi** [comment]
I see in the [source code](https://github.com/clojure/clojurescript/blob/r1.11.60/src/main/clojure/cljs/core.cljc#L2983-L3036) for `require` that it ends up being a `ns*`, and when `*allow-ns*` is falsy that [throws with a "must appear at the top-level"](https://cljs.github.io/api/cljs.core/nsSTAR). I guess in this case `*allow-ns*` is truthy because I don't see such an error. I don't see what's actually happening.

Anyway, setting "cljs" to `null`, as you suggested, worked... meaning that at least the error disappears, even if the require still has to be done manually.

Thanks!

**Comment by PEZ** [comment]
Hello again. I think I fixed the problem. The fix is included in latest Calva, v2.0.404.

Please give it a spin with default `calva.autoEvaluateCode` settings (deleting the entry from your settings json) and let me know.

**Comment by xavi** [comment]
It works perfectly now, thank you!
